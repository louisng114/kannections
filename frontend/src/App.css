#root {
  display: flex;
  min-height: 100vh;
  margin: 0 auto;
  align-items: center;
  text-align: center;
}
