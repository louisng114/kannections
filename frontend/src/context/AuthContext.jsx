import React, { createContext, useState, useEffect } from 'react';
import { jwtDecode } from "jwt-decode";
import KannectionsApi from '../helpers/api';

const AuthContext = createContext();

// functions for user authentication
const AuthProvider = ({ children }) => {
  const [user, setUser] = useState(null);

  useEffect(() => {
    const storedToken = localStorage.getItem('token');
    if (storedToken) {
      KannectionsApi.token = storedToken;
      fetchUserFromToken(storedToken);
    }
  }, []);

  const fetchUserFromToken = async (token) => {
    try {
      const decoded = jwtDecode(token);
      const user = await KannectionsApi.getUser(decoded.username);
      setUser(user);
    } catch (error) {
      console.error('Failed to decode token:', error);
    }
  };

  const signup = async (username, password) => {
    try {
        const token = await KannectionsApi.register({ username, password });
        localStorage.setItem('token', token);
        KannectionsApi.token = token;
        fetchUserFromToken(token);
    } catch (error) {
      console.error("Registration failed:", error);
      throw error;
    }
  };

  const login = async (username, password) => {
    try {
      const token = await KannectionsApi.getToken({ username, password });
      localStorage.setItem('token', token);
      KannectionsApi.token = token;
      fetchUserFromToken(token);
    } catch (error) {
      console.error("Login failed:", error);
      throw error;
    }
  };

  const logout = () => {
    localStorage.removeItem('token');
    setUser(null);
    KannectionsApi.token = null;
  };

  return (
    <AuthContext.Provider value={{ user, signup, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

export { AuthProvider, AuthContext };
