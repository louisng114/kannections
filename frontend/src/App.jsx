import { BrowserRouter, Routes, Route } from 'react-router-dom';
import 'bootstrap/dist/css/bootstrap.min.css';
import "./App.css"
import Navbar from "./components/Navbar";
import Home from "./components/Home";
import GamePage from "./components/GamePage";
import LoginForm from './components/LoginForm';
import SignupForm from './components/SignupForm';
import Profile from './components/Profile';


const App = () => {
    return (
        <BrowserRouter>
            <Navbar />
            <Routes>
                <Route path="/" element={<Home />} />
                <Route path="/game1" element={<GamePage jlpt={1} />} />
                <Route path="/game2" element={<GamePage jlpt={2} />} />
                <Route path="/game3" element={<GamePage jlpt={3} />} />
                <Route path="/game4" element={<GamePage jlpt={4} />} />
                <Route path="/game5" element={<GamePage jlpt={5} />} />
                <Route path="/profile" element={<Profile />} />
                <Route path="/login" element={<LoginForm />} />
                <Route path="/signup" element={<SignupForm />} />
            </Routes>
        </BrowserRouter>
    );
};

export default App;
