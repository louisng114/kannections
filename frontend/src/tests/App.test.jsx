import { render } from '@testing-library/react';
import App from '../App';
import { AuthProvider } from '../context/AuthContext';

test('renders without crashing', () => {
  render(
    <AuthProvider>
      <App />
    </AuthProvider>
  );
});

it("matches snapshot", function() {
  const {asFragment} = render(
      <AuthProvider>
        <App />
      </AuthProvider>
    );
  expect(asFragment()).toMatchSnapshot();
});
