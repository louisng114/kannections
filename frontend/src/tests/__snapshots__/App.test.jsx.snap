// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="bg-body-tertiary navbar navbar-expand-lg navbar-light bg-light fixed-top"
  >
    <div
      class="d-flex justify-content-between container"
    >
      <a
        class="navbar-brand"
        href="/"
      >
        <ruby>
          漢
          <rt>
            KAN
          </rt>
        </ruby>
        nections
      </a>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="mx-auto d-flex align-items-center navbar-nav"
        >
          <div
            class="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="dropdown-toggle nav-link"
              href="#"
              role="button"
              tabindex="0"
            >
              Game
            </a>
          </div>
        </div>
        <div
          class="d-flex align-items-center navbar-nav"
        >
          <a
            class="btn btn-outline-info m-2"
            href="/login"
            role="button"
            tabindex="0"
          >
            Log In
          </a>
          <a
            class="btn btn-outline-success m-2"
            href="/signup"
            role="button"
            tabindex="0"
          >
            Sign Up
          </a>
        </div>
      </div>
    </div>
  </nav>
  <h1>
    WELCOME TO 
    <ruby>
      漢
      <rt>
        KAN
      </rt>
    </ruby>
    NECTIONS!
  </h1>
</DocumentFragment>
`;
